core/src/main/java/org/apache/iceberg/DefaultValueParser.java

```diff
@@ -0,0 +1,363 @@
+        Preconditions.checkArgument(defaultValue.isFloatingPointNumber(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        return defaultValue.floatValue();
+      case DOUBLE:
+        Preconditions.checkArgument(defaultValue.isNumber(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        return defaultValue.doubleValue();
+      case DECIMAL:
+        Preconditions.checkArgument(defaultValue.isNumber(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        BigDecimal retDecimal = defaultValue.decimalValue();
+        Preconditions.checkArgument(
+            retDecimal.scale() == ((Types.DecimalType) type).scale(), "Cannot parse default as a %s value: %s",
+            type,
+            defaultValue);
+        return retDecimal;
+      case STRING:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        return defaultValue.textValue();
+      case UUID:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        try {
+          UUID.fromString(defaultValue.textValue());
+        } catch (IllegalArgumentException e) {
+          throw new IllegalArgumentException(String.format("Cannot parse default as a %s value: %s", type,
+              defaultValue));
+        }
+        return UUID.fromString(defaultValue.textValue());
+      case DATE:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        return DateTimeUtil.isoDateToDays(defaultValue.textValue());
+      case TIME:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        return DateTimeUtil.isoTimeToMicros(defaultValue.textValue());
+      case TIMESTAMP:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        if (((Types.TimestampType) type).shouldAdjustToUTC()) {
+          return DateTimeUtil.isoDateTimeTzToMicros(defaultValue.textValue());
+        } else {
+          return DateTimeUtil.isoDateTimeToMicros(defaultValue.textValue());
+        }
+      case FIXED:
+        Preconditions.checkArgument(
+            defaultValue.isTextual(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        int defaultLength = defaultValue.textValue().length();
+        int fixedLength = ((Types.FixedType) type).length();
+        Preconditions.checkArgument(defaultLength == fixedLength * 2,
+            "Default value %s is not compatible with the expected fixed type, the fixed type is expected to store " +
+                "exactly %s bytes, which means the default value should be of exactly 2 * %s length hex string",
```

Comment:
Can you rephrase this error message? We try to make messages helpful, but short. Instead, this should be "Cannot parse default %s value, incorrect length: %s"

Reply:
fixed